### DISCUSSION TITLE
[Help needed] Add custom Layout

### DISCUSSION THREAD

**Discussion by VasseurLaurent** [vuepress-theme-hope/vuepress-theme-hope#1937] [Q&A] [answered]
Hello everyone ! 

I am currently testing your theme and vuepress for the first time. It is great ! 

However, I have a question, I am trying to create a custom Layout named `Library`
I already read this documentation : [Customize-layout]((https://vuepress-theme-hope.github.io/v2/guide/layout/page.html#customize-layout) , however I don't success to create a Layout that I could call like this : 

```
---
layout: Library
---
```

I also tried things written in this documentation [Vuepress - Extending](https://v2.vuepress.vuejs.org/reference/default-theme/extending.html#layout-slots).

Do you have any tutorial or code that show how to create a new `Layout` from "scratch" ? If no, can you share me some code that shows how to do it, maybe after I could add it to vuepress-theme-hope documentation.

Thank you a lot for your help !

**Comment by Mister-Hope** [comment, marked as answer]
https://vuepress-theme-hope.github.io/v2/cookbook/advanced/extend.html#theme-extending

Plz read the docs carefully, to save time for both you and me. 🙂

**Reply by Mister-Hope** [reply]
Also we improve docs with https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/5308fb225e48eb7db39030955717b6d59ae9b57a

**Reply by VasseurLaurent** [reply]
I didn't see this page before, thank you a lot.
Great idea to add the link to this doc in the `Customize Layout` part 😉
